pkg/scheduler/queue/queue.go

```diff
@@ -437,7 +437,8 @@ func (q *RequestQueue) WaitForRequestForQuerier(ctx context.Context, last Tenant
 		querierConnCtx:  ctx,
 		querierID:       QuerierID(querierID),
 		lastTenantIndex: last,
-		recvChan:        make(chan requestForQuerier),
+		// recvChan is written to by the dispatcher loop and read below. It is never closed.
```

Comment:
let's just move a comment about this down to `send` where it is actually relevant

Reply:
I am actually dubious of adding this comment at all. I don't think it's surprising or notable to leave a channel open. Seeing a channel _closed_ is notable, as I know I'll need to make sure writers and readers are handling the closed channel appropriately. How about I just delete the comment?

Reply:
deleting sounds good to me